ios/FluentUI/Card Nudge/CardNudge.swift

```diff
@@ -40,35 +40,32 @@ public typealias CardNudgeButtonAction = ((_ state: MSFCardNudgeState) -> Void)
 
     /// Action to be dispatched by the dismiss ("close") button on the trailing edge of the control.
     @objc var dismissButtonAction: CardNudgeButtonAction? { get set }
-
-    /// Custom design token set for this control, to use in place of the control's default Fluent tokens.
-    @objc var overrideTokens: CardNudgeTokens? { get set }
 }
 
 /// View that represents the CardNudge.
-public struct CardNudge: View, ConfigurableTokenizedControl {
+public struct CardNudge: View, TokenizedControlView {
+
+    public typealias TokenSetKeyType = CardNudgeTokenSet.Tokens
+    public typealias TokenSetType = CardNudgeTokenSet
+
+    @ObservedObject public var tokenSet: CardNudgeTokenSet = .init()
+
     @Environment(\.fluentTheme) var fluentTheme: FluentTheme
     @ObservedObject var state: MSFCardNudgeStateImpl
-    let defaultTokens: CardNudgeTokens = .init()
-    var tokens: CardNudgeTokens {
-        let tokens = resolvedTokens
-        tokens.style = state.style
-        return tokens
-    }
 
     @ViewBuilder
     var icon: some View {
         if let icon = state.mainIcon {
             ZStack {
-                RoundedRectangle(cornerRadius: tokens.circleRadius)
-                    .frame(width: tokens.circleSize, height: tokens.circleSize)
-                    .foregroundColor(Color(dynamicColor: tokens.buttonBackgroundColor))
+                RoundedRectangle(cornerRadius: tokenSet[.circleRadius].float)
```

Comment:
Mostly for my information, but what is it about this implementation that is making you have to explicitly call out the type of the token like this?

Reply:
Sure! So the short version is that `tokenSet[xyz]` doesn't return a token directly; it returns an enum of type `ControlTokenValue`. Sadly, Swift doesn't support any sort of automatic unboxing, so we need to explicitly extract the raw value of that enum type. To that end I created vars on `ControlTokenValue` for each of the types it potentially represents (`float`, `DynamicColor`, `FontInfo`, and `ShadowInfo`).